samza-api/src/main/java/org/apache/samza/system/SystemFactory.java

```diff
@@ -32,4 +32,16 @@
   SystemProducer getProducer(String systemName, Config config, MetricsRegistry registry);
 
   SystemAdmin getAdmin(String systemName, Config config);
+
+  default SystemConsumer getConsumer(String systemName, Config config, MetricsRegistry registry, String consumerIdPrefix) {
```

Comment:
Why assume that it's used as prefix in the API? Maybe just call this consumerId and let the actual implementation figure out how to use it?

Reply:
I think we don't want caller to assume that this is the id and it must be used as is. The idea is that it should be a unique piece of info that can be used to construct an id. Please suggest your name.

Reply:
I'm assuming this is intended to be used as part of the Kafka clientId to figure out who owns the instance? 

Maybe consumerLabel. We should add some documentation on what this should be set to by the caller (e.g., set this to indicate ownership of the client instance), how this will be used by system implementers (e.g., to identify consumers in logs, threads and client instances etc., along with other relevant information like systemName).

Reply:
by "add some documentation" you mean add  javadoc or open source documentation?

Reply:
this one: https://samza.apache.org/learn/documentation/0.7.0/api/javadocs/org/apache/samza/system/SystemFactory.html ?

Reply:
@MabelYC I mean Javadocs. The page you linked is auto-generated from Javadocs.